bolt/lib/Rewrite/RewriteInstance.cpp

```diff
@@ -2982,6 +2983,9 @@ void RewriteInstance::selectFunctionsToProcess() {
     if (mustSkip(Function))
       return false;
 
+    if (opts::MatchingFunctionsWithHash)
+      return true;
```

Comment:
Since we're forcing the processing of all functions here to construct CFG and compute hashes, let's also mark functions without profile as ignored in Lite mode. 
This should happen after we assign profile, at the end of YAMLProfileReader::readProfile. Under the new option, go over all functions and check if the function has profile, if not, mark it ignored.